mobile/test/common/integration/sds_integration_test.cc

```diff
@@ -16,77 +14,80 @@ namespace {
 
 class SdsIntegrationTest : public XdsIntegrationTest {
 public:
-  SdsIntegrationTest() {
-    override_builder_config_ = true;
-    ExtensionRegistry::registerFactories();
+  void initialize() override {
+    create_xds_upstream_ = false;
+    use_lds_ = false;
     skip_tag_extraction_rule_check_ = true;
-    upstream_tls_ = true;
-
-    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
-      // The default stats config has overenthusiastic filters.
-      bootstrap.clear_stats_config();
-
-      // Add two clusters by default:
-      //  - base: An HTTP2 cluster with one fake upstream endpoint, for accepting requests from
-      //  EM.
-      //  - xds_cluster.lyft.com: An xDS management server cluster, with one fake upstream endpoint.
-      bootstrap.mutable_static_resources()->clear_clusters();
-      bootstrap.mutable_static_resources()->add_clusters()->MergeFrom(
-          createSingleEndpointClusterConfig("base"));
-      bootstrap.mutable_static_resources()->add_clusters()->MergeFrom(
-          createSingleEndpointClusterConfig(std::string(XDS_CLUSTER)));
-    });
-
-    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
-      // Change the base cluster to use SSL and SDS.
-      auto* transport_socket =
-          bootstrap.mutable_static_resources()->mutable_clusters(0)->mutable_transport_socket();
-      envoy::extensions::transport_sockets::tls::v3::UpstreamTlsContext tls_context;
-      tls_context.set_sni("lyft.com");
-      auto* secret_config =
-          tls_context.mutable_common_tls_context()->add_tls_certificate_sds_secret_configs();
-      setUpSdsConfig(secret_config, "client_cert");
-
-      transport_socket->set_name("envoy.transport_sockets.tls");
-      transport_socket->mutable_typed_config()->PackFrom(tls_context);
-    });
+    ExtensionRegistry::registerFactories();
+
+    XdsIntegrationTest::initialize();
+    initializeXdsStream();
   }
 
-  void createUpstreams() override {
-    // This is the fake upstream configured with SSL for the base cluster.
-    addFakeUpstream(Ssl::createUpstreamSslContext(context_manager_, *api_), upstreamProtocol(),
-                    /*autonomous_upstream=*/true);
+  void createEnvoy() override {
+    const std::string target_uri = Network::Test::getLoopbackAddressUrlString(ipVersion());
+    Platform::XdsBuilder xds_builder(target_uri,
+                                     fake_upstreams_.back()->localAddress()->ip()->port());
+    xds_builder.addClusterDiscoveryService();
+    builder_.setXds(std::move(xds_builder));
+    XdsIntegrationTest::createEnvoy();
   }
 
+  void createUpstreams() override { addFakeUpstream(Http::CodecType::HTTP2); }
+
 protected:
+  void sendCdsResponse() {
+    auto cds_cluster = createSingleEndpointClusterConfig(std::string(XDS_CLUSTER_NAME));
+    // Update the cluster to use SSL.
+    auto* transport_socket = cds_cluster.mutable_transport_socket();
```

Comment:
nit: can move this down to right after L52, where it is first used.

Reply:
Done.